File: src/app/api/register/route.js
import { MongoClient } from 'mongodb';

export async function GET(req) {
  console.log("=== REGISTER API CALLED ===");

  // Fix URL parsing
  const url = new URL(req.url, `http://${req.headers.get('host')}`);

  // Get parameters - exactly like your frontend sends
  const fullName = url.searchParams.get('fullName');
  const email = url.searchParams.get('email');
  const password = url.searchParams.get('password');

  console.log("Data received:", { fullName, email, password });

  if (!email || !password) {
    console.log("Missing email or password");
    return Response.json({ Valid: false });
  }

    const client = new MongoClient(
      "mongodb+srv://B00160418:mongo1234@cluster0.jf7grv6.mongodb.net/WealthBeing"
    );

    await client.connect();
    console.log("Connected to DB");

    const db = client.db("WealthBeing");
    const users = db.collection("Login");

    // Check if exists
    const existing = await users.findOne({ email: email });

    if (existing) {
      console.log("Email already exists");
      await client.close();
      return Response.json({ Valid: false });
    }

    // Insert - exactly like your original
    console.log("Inserting new user...");
    const result = await users.insertOne({
      fullName: fullName,
      email: email,
      password: password
    });

    console.log("Insert result:", result);

    await client.close();

    return Response.json({ Valid: true });

  }

File: src/app/api/register/route.py
import os
import bcrypt
from fastapi import APIRouter
from dotenv import load_dotenv
from pymongo import MongoClient

load_dotenv()

router = APIRouter()

client = MongoClient(os.getenv('MONGO_URI'))
db = client[os.getenv('DATABASE_NAME')]

@router.get('/api/register')
def register(fullName: str, email: str, password: str):
    existing = db.Login.find_one({'email': email})

    if existing:
        return {
            'valid': False,
            'message': 'User already exists'
        }

    hashed = bcrypt.hashpw(password.encode(), bcrypt.gensalt())

    db.Login.insert_one({
        'fullName': fullName,
        'email': email,
        'password': hashed
    })

    return {
        'valid': True
    }